edx/edx-enterprise
# ENT-594 Automate Enterprise Data Reporting Configuration

**Description:** This PR introduces a new django app called enterprise_reporting which is meant to hold functionality pertaining to generating and sending reports to enterprise customers without any particular integration point (which would live in integrated_channels). This also introduces the EnterpriseCustomerReportingConfiguration model, which can be created in django admin to specify information needed for sending the report via email to the enterprise customer.

There is also an api endpoint for reading the model that behaves as other read only endpoints for enterprise: <base_lms_url>/enterprise/api/v1/enterprise_customer_reporting

It can be filtered by enterprise customer by appending the query parameter: enterprise_customer=<enterprise_uuid>

**JIRA:** https://openedx.atlassian.net/browse/ENT-594

**Dependencies:** 

**Merge deadline:** 

**Installation instructions:** As expected, but make sure to run migrations.

**Testing instructions:**
There should now be an enterprise reporting section in django admin with one model, Enterprise Customer Reporting Configuration. You should be able to create one linked to an existing enterprise customer. Based on the frequency that is configured, the day_of_month and day_of_week fields will be validated, and set to none if not relevant for the frequency that is configured. The password field is not editable and is generated once the model is saved, at which point it is visible.

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.
